portal/src/main/webapp/WEB-INF/jsp/visualize.jsp

```diff
@@ -85,47 +85,52 @@
 
                 if (values != null && values.length >0)
                 {
-                    for (int i=0; i<values.length; i++)
-                    {
+                    for (int i=0; i<values.length; i++) {
+                    	
+                    	Boolean genesetQuery = false;
+		               	if (localGeneSetList.length() > 0 && localGeneList.length() == 0) {
+                    		genesetQuery = true;
+                    	}
+                    	
                         String currentValue = values[i].trim();
 
-                        if (currentValue.contains("mutation") && !disabledTabs.contains("mutations"))
+                        if ((currentValue.contains("mutation") && !disabledTabs.contains("mutations")) && !genesetQuery)
```

Comment:
@zhx828 I have rewritten the code. I think now looks nicer, thank you for your suggestion.